### PR TITLE
feat: accounts profile integration

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Able to update the profile information in API service.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

**Review by rebelchris — CHANGES_REQUESTED** [review]
@sshanzel I'm pretty sure we only need to update the API for the profile information.
Only the email change (separate flow) should notify both.

File: packages/shared/src/lib/auth.ts [hunk 1]
```diff
@@ -295,6 +307,42 @@ export const logoutSession = (
     headers: { Accept: 'application/json', 'X-CSRF-Token': csrf_token },
   });
 
+export const initializeSettings = async (): Promise<InitializationData> => {
```
Comment (rebelchris) [inline packages/shared/src/lib/auth.ts:310]: Do we still need this?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice, quick fix 🙏

Just one note on the settings, unsure if we still need it?

File: packages/shared/src/lib/auth.ts (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/lib/auth.ts:310]: Should be not, I think. Let me remove this one quickly! Thanks for the fast review! 🚢

File: packages/shared/src/components/auth/RegistrationForm.tsx [hunk 2]
```diff
@@ -84,7 +84,9 @@ export const RegistrationForm = ({
         ref={formRef}
         onSubmit={onSubmit}
       >
-        {socialAccount && <ImageInput initialValue={socialAccount.image} />}
+        {socialAccount && (
```
Comment (rebelchris) [inline packages/shared/src/components/auth/RegistrationForm.tsx:87]: We need to ensure that this is non-editable for the social flow.
See WT-348, (just a reminder if you can pick that one up after this?)

**Review by rebelchris — APPROVED** [review]
Awesome work, really happy with this one so far

File: packages/shared/src/components/auth/RegistrationForm.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/auth/RegistrationForm.tsx:87]: Would be more than happy to take it 🚢

File: packages/shared/src/components/auth/AuthOptions.tsx [hunk 3]
```diff
@@ -64,13 +64,9 @@ function AuthOptions({
...
     hasLoggedOut() ? Display.SignBack : defaultDisplay,
   );
-  const { onUpdateSession } = useContext(AuthContext);
   const { validateRegistration, onSocialRegistration } = useRegistration({
     key: 'registration_form',
-    onValidRegistration: (data) => {
-      onUpdateSession(data);
-      setActiveDisplay(Display.EmailSent);
-    },
+    onValidRegistration: () => setActiveDisplay(Display.EmailSent), // on valid registration get boot
```
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthOptions.tsx:69]: Are we stilling missing this boot call?

File: packages/shared/src/components/auth/LoginForm.tsx [hunk 4]
```diff
@@ -40,12 +37,7 @@ function LoginForm({
   );
   const { mutateAsync: onPasswordLogin } = useMutation(
     (params: ValidateLoginParams) => submitKratosFlow(params),
-    {
-      onSuccess: ({ data }) => {
-        const session = data as AuthSession;
-        onUpdateSession(session);
-      },
-    },
+    // { onSuccess: ({ data }) => {} }, // refetch boot after login
```
Comment (rebelchris) [inline packages/shared/src/components/auth/LoginForm.tsx:40]: Same here, are you tackling this seperate?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Almost there 👀

File: packages/shared/src/components/auth/LoginForm.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/components/auth/LoginForm.tsx:40]: Yes, it will be part of the held off PR for login and registration.

**Pull request merged**